File: ASP.NET Web Forms Application Using Entity Framework Database First - V2/[C#]-ASP.NET Web Forms Application Using Entity Framework Database First - V2/C#/ContosoUniversity/DAL/ISchoolRepository.cs
using System;
using System.Collections.Generic;

namespace ContosoUniversity.DAL
{
    public interface ISchoolRepository : IDisposable
    {
        IEnumerable<Department> GetDepartments();
        IEnumerable<Department> GetDepartments(string sortExpression);
        IEnumerable<Department> GetDepartmentsByName(string sortExpression, string nameSearchString);
        IEnumerable<Department> GetDepartmentsByAdministrator(Int32 administrator);
        void InsertDepartment(Department department);
        void DeleteDepartment(Department department);
        void UpdateDepartment(Department department, Department origDepartment);
        IEnumerable<InstructorName> GetInstructorNames();
        IEnumerable<OfficeAssignment> GetOfficeAssignments(string sortExpression);
        void InsertOfficeAssignment(OfficeAssignment OfficeAssignment);
        void DeleteOfficeAssignment(OfficeAssignment OfficeAssignment);
        void UpdateOfficeAssignment(OfficeAssignment OfficeAssignment, OfficeAssignment origOfficeAssignment);
    }
}

File: ASP.NET Web Forms Application Using Entity Framework Database First - V2/[VB]-ASP.NET Web Forms Application Using Entity Framework Database First - V2/VB/ContosoUniversity/DAL/ISchoolRepository.vb
Public Interface ISchoolRepository
    Inherits IDisposable

    Function GetDepartments() As IEnumerable(Of Department)
    Function GetDepartments(ByVal sortExpression As String) As IEnumerable(Of Department)
    Function GetDepartmentsByName(ByVal sortExpression As String, ByVal nameSearchString As String) As IEnumerable(Of Department)
    Function GetDepartmentsByAdministrator(ByVal administrator As Int32) As IEnumerable(Of Department)
    Sub InsertDepartment(ByVal department As Department)
    Sub DeleteDepartment(ByVal department As Department)
    Sub UpdateDepartment(ByVal department As Department, ByVal origDepartment As Department)
    Function GetInstructorNames() As IEnumerable(Of InstructorName)
    Function GetOfficeAssignments(ByVal sortExpression As String) As IEnumerable(Of OfficeAssignment)
    Sub InsertOfficeAssignment(ByVal OfficeAssignment As OfficeAssignment)
    Sub DeleteOfficeAssignment(ByVal OfficeAssignment As OfficeAssignment)
    Sub UpdateOfficeAssignment(ByVal OfficeAssignment As OfficeAssignment, ByVal origOfficeAssignment As OfficeAssignment)
End Interface
